src/EditorFeatures/CSharpTest/AddBraces/AddBracesTests.cs

```diff
@@ -368,13 +648,120 @@ static void Main()
             return;
         }
     }
-}");
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, true)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        public async Task FireForMultilineForWithoutBraces1(int bracesPreference, bool expectDiagnostic)
+        {
+            await TestMissingOrInRegularAndScriptAsync(
+            @"
+class Program
+{
+    static void Main()
+    {
+        [|for|] (var i = 0;
+            i < 5;
+            i++) return;
+    }
+}",
+
+   @"
+class Program
+{
+    static void Main()
+    {
+        for (var i = 0;
+            i < 5;
+            i++)
+        {
+            return;
+        }
+    }
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, true)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        public async Task FireForMultilineForWithoutBraces2(int bracesPreference, bool expectDiagnostic)
+        {
+            await TestMissingOrInRegularAndScriptAsync(
+            @"
+class Program
+{
+    static void Main()
+    {
+        [|for|] (var i = 0; i < 5; i++) if (true)
+            return;
+    }
+}",
+
+   @"
+class Program
+{
+    static void Main()
+    {
+        for (var i = 0; i < 5; i++)
+        {
+            if (true)
+            return;
+        }
+    }
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
         }
 
-        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
-        public async Task FireForForEachWithoutBraces()
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, true)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        public async Task FireForMultilineForWithoutBraces3(int bracesPreference, bool expectDiagnostic)
         {
-            await TestInRegularAndScriptAsync(
+            await TestMissingOrInRegularAndScriptAsync(
+            @"
+class Program
+{
+    static void Main()
+    {
+        [|for|] (var i = 0; i < 5; i++)
+            if (true)
+                return;
```

Comment:
i'm trying to reconcile this and FireForIfNestedInElseWithoutBracesWithMultilineContext1.  I'm finding it a little unclear why this should have braces when multiline, but hte linked test shuld not.

Reply:
Braces will be preferred for the top level `if` statement, and after they are added braces will start to be offered for the `else if`. The upside is the checks are more efficient. The downside is fix all may require more than one pass in some very rare edge cases.

Reply:
➡️ Hoping the new documentation and extra tests keep this case clear.